### PR TITLE
add userService with CRUD and change userModel interface

### LINKED ISSUE #9: Create userService with all CRUD operations
https://pl.wikipedia.org/wiki/CRUD
https://coursework.vschool.io/mongoose-crud/

**start this task only when #7 is done**

### PR DESCRIPTION
**Pull request by Sienkowski99**
closes #9

### CODE REVIEW

File: src/services/userService.ts [hunk 1]
```diff
@@ -0,0 +1,52 @@
...
+export class userService {
+  public static addUser = async (data: IUser): Promise<IUser> => {
+    const newUser = new User({
+      registrationDate: new Date(),
+      ...data,
+    });
+    return new Promise(async (resolve, reject) => {
+      newUser
+        .save()
+        .then((res) => resolve(res))
```
Comment (bartlomiejmont) [inline src/services/userService.ts:13]: We want to resolve newUser

File: src/services/userService.ts [hunk 2]
```diff
@@ -0,0 +1,52 @@
...
+  public static deleteUser = async (
+    userID: mongoose.Types.ObjectId
+  ): Promise<string> => {
+    return new Promise(async (resolve, reject) => {
+      User.findOneAndDelete({ _id: userID })
+        .then((res) => resolve(`Removed user with ID ${userID}`))
+        .catch((err) =>
+          reject(
+            `En ERROR has occurred while trying to delete a user with ID ${userID}`
+          )
```
Comment (bartlomiejmont) [inline src/services/userService.ts:48]: res and err is never used

File: src/services/userService.ts [hunk 3]
```diff
@@ -0,0 +1,52 @@
...
+        .catch((err) => reject(err));
+    });
+  };
+
+  public static updateUser = async (
+    data: IUser,
+    userID: mongoose.Types.ObjectId
+  ): Promise<IUser> => {
+    return new Promise(async (resolve, reject) => {
+      User.findOneAndUpdate({ _id: userID }, { $set: data }, { new: true })
```
Comment (bartlomiejmont) [inline src/services/userService.ts:33]: why did you use `{ $set: data }` insted of only `data` - just curious

**Review by bartlomiejmont — CHANGES_REQUESTED** [review]
fix all res res functions https://www.w3schools.com/js/js_arrow_function.asp

File: src/services/userService.ts [hunk 4]
```diff
@@ -0,0 +1,52 @@
...
+    });
+    return new Promise(async (resolve, reject) => {
+      newUser
+        .save()
+        .then((res) => resolve(res))
+        .catch((err) => reject(err));
+    });
+  };
+
+  public static readUser = async (
```
Comment (bartlomiejmont) [inline src/services/userService.ts:18]: getUserById

File: src/services/userService.ts (hunk 3 above)
Comment (Sienkowski99) [reply src/services/userService.ts:33]: Hmm.. I think that I saw it once and it got stuck inside my head. For me it is easier to see what's going on

File: src/services/userService.ts (hunk 1 above)
Comment (Sienkowski99) [reply src/services/userService.ts:13]: changed

File: src/services/userService.ts (hunk 2 above)
Comment (Sienkowski99) [reply src/services/userService.ts:48]: changed

File: src/services/userService.ts (hunk 4 above)
Comment (Sienkowski99) [reply src/services/userService.ts:18]: changed

**Review by bartlomiejmont — APPROVED** [review]

**Pull request merged**